src/rgw/rgw_process.cc

```diff
@@ -312,6 +313,22 @@ int process_request(rgw::sal::RGWRadosStore* const store,
     handler->put_op(op);
   rest->put_handler(handler);
 
+  // access log line elements begin per Apache Combined Log Format with additions following
+  static bool beast_framework = s->cct->_conf->rgw_frontends.find("beast") != string::npos;
+  if ( beast_framework ) {
```

Comment:
i think it would be preferable to move this code out of the generic process_request() code path, and into the beast frontend itself. the `rgw_frontends` configuration is very flexible, and would allow you to run multiple frontends (ie beast and civetweb simultaneously). there isn't a reliable way to detect which frontend issued the request at this point